tools/ci/prepare-pull-credentials.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -e
+
+if [[ "$DOCKERHUB_PULL_CREDENTIAL" == "" ]]; then
+    >&2 echo "No docker pull credential information specified"
+else
+    DOCKER_USER=$(echo "$DOCKERHUB_PULL_CREDENTIAL" | cut -d ':' -f 1)
+    DOCKER_PWD=$(echo "$DOCKERHUB_PULL_CREDENTIAL" | cut -d ':' -f 2)
+    echo -n "$DOCKER_PWD" | docker login -u "$DOCKER_USER" --password-stdin > /dev/null
+    cat <<EOF > "$HOME/.docker/k3d-registry.yaml"
+configs:
+  registry-1.docker.io:
+    auth:
+      username: $DOCKER_USER
+      password: $DOCKER_PWD
+EOF
+    echo "$HOME/.docker/k3d-registry.yaml"
+
+    echo "{\"auths\":{\"https://index.docker.io/v1/\":{\"auth\":\"$CRED\"}}}" > "$HOME/.docker/kind-config.json"
+    EXTRA_MOUNTS="[ {\"containerPath\": \"/var/lib/kubelet/config.json\", \"hostPath\": \"$HOME/.docker/kind-config.json\"}]"
```

Comment:
Could we just have always the same config with an ignored file? Then your make target would create the file it needs systematically (with or without creds). This way using creds is not more complex and there are a lot less conditionals